### ISSUE TITLE
Example make targets and docs don't work well when CRI-O is used as the container runtime

### ISSUE THREAD

**Issue by rktidwell** [networkservicemesh/examples#87]
I was attempting to demo some NSM things to some colleagues. The systems I was working on don't use docker as the container runtime, they use CRI-O. Here's an example of what happens when you have CRI-O instead of docker:

`suse@tidwellr-dev-1:~/examples> make k8s-icmp-save k8s-icmp-load-images
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.40/build?buildargs=%7B%22VPP_AGENT%22%3A%22artembelov%2Fvpp-agent%3Av2.5.1%22%7D&cachefrom=%5B%5D&cgroupparent=&cpuperiod=0&cpuquota=0&cpusetcpus=&cpusetmems=&cpushares=0&dockerfile=examples%2Fproxy%2Fsidecar-nse%2FDockerfile&labels=%7B%7D&memory=0&memswap=0&networkmode=host&rm=1&session=v7sut2ftomcncr9un32rtyctm&shmsize=0&t=networkservicemesh%2Fproxy-sidecar-nse&target=&ulimits=null&version=1: dial unix /var/run/docker.sock: connect: permission denied
make: *** [/home/suse/examples/mk/docker-targets.mk:53: docker-proxy-sidecar-nse-build] Error 1`

I encountered this is on openSUSE Tumbleweed running Kubic. Docker is not installed or running in this configuration. The make scripts here assume `docker build`. Adding support for buildah would be a nice touch. If possible, a little massaging of make targets and docs would be a nice touch for environments where docker is not the container runtime.

**Comment by nickolaev** [comment]
Yep, I can confirm, building images is bound to docker today. But the images are actually available in the docker hub so you should be able to deploy (with proper tags set) only the YAMLs and get the images downloaded for you.

Of course, PRs to support other build tools are welcome :)
